src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -125,7 +125,7 @@ public function collect(Request $request, Response $response, \Exception $except
                 continue;
             }
             if ('request_headers' === $key || 'response_headers' === $key) {
-                $value = array_map(function ($v) { return isset($v[1]) ? $v : $v[0]; }, $value);
+                $value = array_map(function ($v) { return isset($v[1]) ? $v : (isset($v[0]) ? $v[0] : null); }, $value);
```

Comment:
To be even more defensive, I'd suggest: `isset($v[0]) && !isset($v[1]) ? $v[0] : $v`